### DISCUSSION TITLE
TShock 5.2.1 for Terraria 1.4.4.9

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#3044] [Announcements]
This release fixes GHSA-hvm9-wc8j-mgrc (reported by: @sgkoishi, found by: @THEXN).

This also releases the following changes:

## Notable changes

* Updated `TSPlayer.GodMode`. (@AgaSpace)
  * Previously the field was used as some kind of dataset changed by /godmode command, but now it is a property that receives/changes data in journey mode.
* Added the `TSPlayer.Client` property. It allows the developer to get the `RemoteClient` player, without an additional call to `Terraria.Netplay.Clients`. (@AgaSpace)
* Updated the documentation for the `TSPlayer.SetPvP` method. The `sendMsg` parameter, which is responsible for sending a pvp mode change message, was not documented earlier. (@AgaSpace)
* Added methods `TSPlayer.KillPlayer` and `TSPlayer.DamagePlayer` for which you can specify the cause (`PlayerDeathReason`) in parameters. (@AgaSpace)
* Added an error when trying to change a `TSPlayer` team to, say, 9, when there are only 6. (@AgaSpace)
* Added an error when trying to call the `TSPlayer.SetTeam` method with an argument (team) greater than 5 or less than 0. (@AgaSpace)
* Added a method `TSPlayer.UpdateSection` with arguments `rectangle` and `isLoaded`, which will load some area from the server to the player. (@AgaSpace)
* Added a method `TSPlayer.GiveItem`, which has `TShockAPI.NetItem` structure in its arguments. (@AgaSpace)
* Added a property `TSPlayer.Hostile`, which gets pvp player mode. (@AgaSpace)
* Fixed bug where when the `UseSqlLogs` config property is true, an empty log file would still get created. (@ZakFahey)
* Fixed typo in `/gbuff`. (@sgkoishi, #2955)
* Rewrote the `.dockerignore` file into a denylist. (@timschumi)
* Added CI for Docker images. (@timschumi)
* Fixed Cursed Flares kicking players for invalid buff. (@Arthri)
## Upgrading

**Before installing, make a complete backup of your existing server and all files and folders, including `tshock.sqlite` and the `tshock` folder. We are not responsible for failed installations. Please make sure you backup your data. Severe data loss or gain may occur.**

To upgrade TShock, replace all of the files in the location of your server installation with all of the files from the release zip we provide. Overwrite all files. No data will be lost.

<hr /><em>This discussion was created from the release <a href='https://github.com/Pryaxis/TShock/releases/tag/v5.2.1'>TShock 5.2.1 for Terraria 1.4.4.9</a>.</em>

**Comment by AREALBUILDER** [comment]
#ЗУМ_ВЕРНИСЬ_НА_ТЗ #ТЗ_НУЖДАЕТСЯ_В_ПЕРЕМЕНАХ
Я понимаю, что будет сложно вернуться, но вернуться надо.

**Reply by AgaSpace** [reply]
> #ЗУМ_ВЕРНИСЬ_НА_ТЗ #ТЗ_НУЖДАЕТСЯ_В_ПЕРЕМЕНАХ
> Я понимаю, что будет сложно вернуться, но вернуться надо.

Dont write this here. You can dm me in discord

**Reply by hakusaro** [reply]
lol

**Reply by AREALBUILDER** [reply]
> > #ЗУМ_ВЕРНИСЬ_НА_ТЗ #ТЗ_НУЖДАЕТСЯ_В_ПЕРЕМЕНАХ 
> > Я понимаю, что будет сложно вернуться, но вернуться надо.
> 
> Не пиши это здесь. Можешь написать мне в дискорде
Для начала прими в друзья

**Comment by CristopherASC** [comment]
soooo bad D: , I was excited but your file marks viruz

**Reply by hakusaro** [reply]
Your antivirus software has a false positive. Call them, not us.

**Reply by CristopherASC** [reply]
If I was just seeing what the file does through a virtual machine, even if I didn't leave it for long, I was checking to see if it is a false positive, it seems to be a legitimate file

**Comment by CarlucciMods** [comment]
Is this coming to Pterodactyl Panel, Nest Egg ?
